### PR TITLE
feat: donation redesign (migrate db schema & update api endpoint)

### PR DESCRIPTION
**Pull request by Aylie-Chou**
- [TWREPORTER-209](https://twreporter-org.atlassian.net/browse/TWREPORTER-209)
  - add migrate up / down file
- [TWREPORTER-217](https://twreporter-org.atlassian.net/browse/TWREPORTER-217)
  - update api endpoint

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,17 +47,25 @@ create-migrations: check-migrate
 		@migrate create -ext $(MIGRATION_EXT) -dir $(MIGRATION_DIR) -seq $(MIGRATION_NAME)
 
 upgrade-schema: check-migrate
-		@echo Upgrade schema to latest version
-		migrate -database $(DB_CONN) -path $(MIGRATION_DIR) up
+		@echo Upgrade UP schema or to latest version
+		migrate -database $(DB_CONN) -path $(MIGRATION_DIR) up $(UP)
```
Comment (taylrj) [inline Makefile:51]: Question: do we need to define `UP` or `DOWN` ?

Comment (Aylie-Chou) [reply Makefile:51]: We can define `UP` or `DOWN` using variable assign like following:
```
// upgrade one version
$ make upgrade-schema UP=1

// if UP is undefined, upgrade to latest version
$ make upgrade-schema
```

Comment (taylrj) [reply Makefile:51]: After a second thought, the naming `UP` and `DOWN` looks good to me.

Could you help to add the comments about using variable assign as above into README.md [migration section](https://github.com/twreporter/go-api#database-migrations) ?

Comment (Aylie-Chou) [reply Makefile:51]: updated [1831ab3](https://github.com/twreporter/go-api/pull/463/commits/1831ab39a80e350600cc69d5afc2f76021b7d641)

File: tests/controller_donations_test.go [hunk 2]
```diff
@@ -55,17 +55,16 @@ type (
...
-		Cardholder  models.Cardholder `json:"donor"`
-		Currency    string            `json:"currency"`
-		Details     string            `json:"details"`
-		Frequency   string            `json:"frequency"`
-		MerchantID  string            `json:"merchant_id"`
-		PayMethod   string            `json:"pay_method"`
-		Prime       string            `json:"prime"`
-		UserID      uint              `json:"user_id"`
-		ToFeedback  bool              `json:"to_feedback"`
-		IsAnonymous bool              `json:"is_anonymous"`
```
[the remark this answers is not in the document]
Comment (Aylie-Chou) [reply tests/controller_donations_test.go:68]: requestBody is for creating default donation, which do not contain donor info like IsAnonymous or ToFeedback.

File: controllers/donation.go [hunk 3]
```diff
@@ -755,9 +758,8 @@ func (mc *MembershipController) PatchADonationOfAUser(c *gin.Context, donationTy
 		}, err
 	}
 
-	if err = mc.UpdateUserDataByCardholder(_cardholder, userID); err!= nil {
-		fmt.Println("sync data to user table err: %+v", err)
-	}
+	// sync user data by cardholder aynchronous
+	go mc.UpdateUserDataByCardholder(_cardholder, userID)
```
[the remark this answers is not in the document]
Comment (Aylie-Chou) [reply controllers/donation.go:762]: update. [4d95717](https://github.com/twreporter/go-api/pull/463/commits/4d9571758d505fb6ea89ad1e0e8246a238b68268)

**Review by taylrj — APPROVED** [review]
LGTM 👍

**Pull request merged**